qiskit/circuit/quantumcircuit.py

```diff
@@ -2724,6 +2724,8 @@ def append(
             if copy and is_parameter:
```

Comment:
Above this line (beyond the scope of your changes), I see that we assign to `param`. My eyes might be playing tricks on me, but is there a point to that?

Reply:
I don't know why we bother with the assignment in the lines above - the commit that introduced the assignment (#10977) doesn't say why - before, we didn't have the assignment -, and we drop the variable immediately after.  I suspect `_validate_expr` returns the input just for chaining convenience - there's at least one place that does `node = _validate_expr(expr.lift(node))`, so that's probably the base.

Either that, or some earlier version of the code (as in, during local development) had more of a call for it.